processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -63,16 +64,21 @@ type WatchClient struct {
 // format: [deployment-name]-[Random-String-For-ReplicaSet]-[Random-String-For-Pod]
 var dRegex = regexp.MustCompile(`^(.*)-[0-9a-zA-Z]*-[0-9a-zA-Z]*$`)
 
+// Extract statefulset name from the pod name. Pod name is created using
+// format: [statefulset-name]-[pod-number]
+var stsRegex = regexp.MustCompile(`^(.*)-[0-9]*$`)
```

Comment:
There is an issue with the approach, even for getting `k8s.deployment.name`. Potentially this can match a standalone pod with a name like `test-pod-1` or even a regular deamonset pod with a very low chance. Ideally we should use k8s API to fetch owner of the pod and set `k8s.statefulset.name` if applicable.

Reply:
Submitted an issue for `k8s.deployment.name` https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/12936.

I'm not sure we want to apply the same not ideal approach for `k8s.statefulset.name